### ISSUE TITLE
Bug in navigation by letter in the media library

### ISSUE THREAD

**Issue by Bindou01** [xbmc/xbmc#25921]
### Describe the bug

I notice an unexpected behavior when you put "Title" as the sort method and use the page selector. At the beginning you have to click 20 times to move forward one page, hahaha! 
-> [https://www.youtube.com/watch?v=MVExUEbwK_0](https://github.com/xbmc/xbmc/issues/url)

## Expected Behavior

The selector should follow the following order:
1->A->B->C->D etc.

## Actual Behavior

When you move from one page to another, it starts by moving the numbers forward, but it does so in an anarchic order. We see the numbers moving around : 2->3->6->8->1->4->6->1->2
Then it gets to the letter A, and reviews all its variants (some even come back several times) : A->À->A->À->A->Â-> and only then do we get to the letter B.
And there is the same problem with the letter E.

### To Reproduce

Open a skin that has a page selector by letter.
Open a media library that has a lot of content.
Move the selector, and observe.

### Debuglog

The debuglog can be found here:

### Screenshots 
Here are some links or screenshots to help explain the problem:

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

![image](https://github.com/user-attachments/assets/c4e33c10-a4c8-4a00-a117-18b02ab7f64a)

### Your Environment
Used Operating system:

 - [X] Android
 - [ ] iOS
 - [ ] tvOS
 - [X] Linux
 - [ ] macOS
 - [X] Windows
 - [ ] Windows UWP

 - Operating system version/name: Windows 11 and CoreElec
 - Kodi version: 21.1

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by Hitcher** [comment]
The page movement isn't per letter, it's per page of items on screen. So in your screenshot it will move 12 items at a time.

**Comment by Bindou01** [comment]
> The page movement isn't per letter, it's per page of items on screen. So in your screenshot it will move 12 items at a time.

In this case, the movement is done by letter, and no element moves before reaching the letter B. So I don't understand your answer.

**Comment by Hitcher** [comment]
The letter is just telling you the first letter of the currently selected item.

The scrollbar is only moving a page of items at a time.

![screenshot00012](https://github.com/user-attachments/assets/5bcacfcb-f783-4f15-8943-2556bfc33428)

![screenshot00013](https://github.com/user-attachments/assets/099c0055-a3a9-43c8-905c-18859d0fd719)

**Comment by Bindou01** [comment]
Please, Hitcher, take the time to watch the video that illustrates the problem:

[https://www.youtube.com/watch?v=MVExUEbwK_0](url)

Recent skins like Jurial's allow movement by letter, which could be very handy if Kodi didn't react that way. But more like this: 1->A->B->C->D etc.

**Comment by Hitcher** [comment]
Sorry, didn't realise the scrollbar was jumping by letter and not page.

Watched the video and I don't see the problem though as it's jumping correctly through your movies.

**2** Guns
**3** Billboards
**3**:10 To Yuma
**6** Underground
**8** Mile
**12** Hommes en Colère
**13** Hours
**48** Hours
**65** La Terre D'Avant
**101** Dalmations
etc

It's just the the first digit is being displayed in the heading.

If you want the scrollbar to actually scroll through the pages see if there's an option in the skin to change it.

**Comment by Bindou01** [comment]
That's a good point, Hitcher, for the numbers, there is indeed a problem with the skin display.

On the other hand, it doesn't necessarily seem obvious to me to go through all the letters A, à, â, etc. in review, which causes navigation problems like here. A problem that must represent 99% of situations. It would seem more logical to me to go from A to B, right?

But I understood that the system went through all the first letters of the titles present in the library and that this was considered normal behavior. So I would submit to it.

**Comment by Bindou01** [comment]
Also, as Hitcher points out here, Fuse 2 doesn't display numbers. There's only one number displayed instead of two.

https://github.com/xbmc/xbmc/issues/25921

**Comment by Hitcher** [comment]
I'll reopen so hopefully someone who knows the 'jump by letter' logic can reply.

**Comment by Hitcher** [comment]
This seems to be the code that controls it:

https://github.com/xbmc/xbmc/blob/17ac1bbda23b5b292bc1e898404f2e7d165226c9/xbmc/guilib/GUIBaseContainer.cpp#L659
